README.md

```diff
@@ -59,20 +59,34 @@ If you want to automatically search root modules in your workspace and exclude s
 ### Formatting
 To enable formatting, it is recommended that the following be added to the extension settings for the Terraform extension:
 ```
-    "[terraform]": {
-        "editor.formatOnSave": true
-    }
+"[terraform]": {
+  "editor.defaultFormatter": "hashicorp.terraform",
+  "editor.formatOnSave": true,
+  "editor.formatOnSaveMode": "file"
+}
 ```
 
-It is also recommended to set a default formatter if you have other extensions installed which also claim a capability to format Terraform files:
+It is recommended to set `editor.defaultFormatter` to ensure that VS Code knows which extension to use to format your files. It is possible to have more than one extension installed which claim a capability to format Terraform files.
+
+When using the `editor.formatOnSaveMode` setting, only `file` is currently supported. The `modifications` or `modificationsIfAvailable` settings [use the currently configured SCM](https://code.visualstudio.com/updates/v1_49#_only-format-modified-text) to detect file line ranges that have changed and send those ranges to the formatter. The `file` setting works because `terraform fmt` was written to format an entire file, not ranges. If you don't have a SCM enabled for the files you are editing, `modifications` won't work at all. The `modificationsIfAvailable` setting will fall back to `file` if there is no SCM and will appear to work sometimes.
+
+If you want to use `editor.codeActionsOnSave` with `editor.formatOnSave` to automatically format Terraform files, use the following configuration:
 
 ```
-    "[terraform]": {
-        "editor.defaultFormatter": "hashicorp.terraform",
-        "editor.formatOnSave": true
-    }
+"editor.formatOnSave": true,
+"[terraform]": {
```

Comment:
Does this represent language ID? If so, then do we need a similar block for `[terraform-vars]`?

Reply:
Yes, that's the language ID

![image](https://user-images.githubusercontent.com/272569/148390263-236003fc-65f0-4e42-878e-4c56402e8123.png)

Does terraform format tfvars files?

Reply:
It does.

```
Usage: terraform [global options] fmt [options] [DIR]

	Rewrites all Terraform configuration files to a canonical format. Both
	configuration files (.tf) and variables files (.tfvars) are updated.
	JSON files (.tf.json or .tfvars.json) are not modified.
```
> Both configuration files (.tf) and **variables files (.tfvars)** are updated.

Reply:
It's not a big deal I suppose. If we want it to be complete then I suppose we should include it.

Feel free to merge as is though.

Reply:
Might as well since we're here! Added